### PR TITLE
GCP Firestore - Certification Tests 

### PR DESCRIPTION
**Pull request by robertojrojas**
# Description

Provides the GCP Firestore Conformance and Certification Test Plan and Test Code.

## Issue reference

This PR will close: #2460

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests

### CODE REVIEW

File: .github/scripts/test-info.mjs
Comment (artursouza) [inline .github/scripts/test-info.mjs:584]: Do we need destroy scripts for both certification and conformance?

**Review by artursouza — CHANGES_REQUESTED** [review]

File: .github/scripts/test-info.mjs
Comment (robertojrojas) [reply .github/scripts/test-info.mjs:584]: `GPC Firestore`setup with **datastore mode** is really awkward. We have to manually configure it through the Google Cloud console.

File: tests/certification/state/gcp/firestore/gcp_firestore_test.go
Comment (robertojrojas) [reply tests/certification/state/gcp/firestore/gcp_firestore_test.go:1]: I forgot to add one. Adding one now.

File: .github/scripts/test-info.mjs
Comment (berndverst) [inline .github/scripts/test-info.mjs:588]: I assume this should say `requireGCPCredentials`

File: tests/certification/state/gcp/firestore/components/entity_kind/gcpfirestore.yaml
Comment (berndverst) [inline tests/certification/state/gcp/firestore/components/entity_kind/gcpfirestore.yaml:11]: It seems this environment variable isn't getting set correctly.

File: .github/scripts/components-scripts/conformance-state.gcp.firestore-setup.sh
Comment (berndverst) [inline .github/scripts/components-scripts/conformance-state.gcp.firestore-setup.sh:6]: is $GCP_PROJECT actually set at this point?

Comment (robertojrojas) [reply .github/scripts/components-scripts/conformance-state.gcp.firestore-setup.sh:6]: This env var is set by the **Authenticate to Google Cloud** step.

**Comment by berndverst** [comment]
FYI: Due to severe limitations on GitHub runners, but also to safeguard our GitHub secrets only select contributors - maintainers and approvers can run `/ok-to-test`

We can add you in the future after you make some more contributions other than cert tests.

**Comment by artursouza** [comment]
> FYI: Due to severe limitations on GitHub runners, but also to safeguard our GitHub secrets only select contributors - maintainers and approvers can run `/ok-to-test`
> 
> We can add you in the future after you make some more contributions to contrib other than cert tests / conformance tests.

I approve Roberto to have access to /ok-to-test

File: tests/certification/state/gcp/firestore/README.md
Comment (artursouza) [inline tests/certification/state/gcp/firestore/README.md:1]: Thanks.

**Review by artursouza — APPROVED** [review]

**Comment by yaron2** [comment]
> > FYI: Due to severe limitations on GitHub runners, but also to safeguard our GitHub secrets only select contributors - maintainers and approvers can run `/ok-to-test`
> > We can add you in the future after you make some more contributions to contrib other than cert tests / conformance tests.
> 
> I approve Roberto to have access to /ok-to-test. The "select contributors" is mostly about trust and not to have an unknown person to have access to the credentials we use for our tests. I believe Roberto has made the bar as a trusted contributor.

Being an org members should be accounted as prerequisite IMO and non-org members should not have access to build/test checks. For Roberto, I agree he made the bar, also being a Dapr org members.

**Comment by berndverst** [comment]
> 

It's select due to legacy reasons (not because I want this :p) - I think it's better if we restrict it to approvers / maintainers because we do not have the GitHub runner capacity for many people to frequently run these tests.

**Review by berndverst — CHANGES_REQUESTED** [review]
Firestore conformance test for cloud is still failing.

Blocking this PR until the cloud based conformance and certification tests pass.

**Review by artursouza — APPROVED** [review]

**Pull request merged**
